common/lib/src/realm_types.dart

```diff
@@ -224,3 +225,223 @@ class RealmValue {
   @override
   String toString() => 'RealmValue($value)';
 }
+
+/// A base type for the supported geospatial shapes.
+sealed class GeoShape {}
+
+/// A point on the earth's surface.
+///
+/// It cannot be persisted as a property on a realm object.
+///
+/// Instead, you must use a custom embedded object with the following structure:
+/// ```dart
+/// @RealmModel(ObjectType.embeddedObject)
+/// class _Location {
+///   final String type = 'Point';
+///   List<double> coordinates = const [0, 0];
+/// }
```

Comment:
Does the dart generator support playing around with the accessibility of fields? If so, we could suggest a more user-friendly API, similar to what we do in .NET: https://github.com/realm/realm-dotnet/blob/641f2dc5581776b80979c69abc4c29777313bd0b/CHANGELOG.md?plain=1#L140-L162

Reply:
I have added getters and setters for latitude and longitude, as well conversion methods between `GeoPoint` and `Location` in the doc comments (similar to how I did in `geospatial_test.dart`)